### PR TITLE
Add checkUrl before NewWindow

### PR DESCRIPTION
**Pull request by francisbouvier**
Signed-off-by: Francis Bouvier <francis.bouvier@xerus-technologies.fr>

### CODE REVIEW

**Comment by francisbouvier** [comment]
Hi @miketheprogrammer ,

A small PR in order to perform a quick check on the url parameter of the NewWindow function.

If the url scheme is empty this helper try to load the file from current directory.
So you can do:
```go
thrust.NewWindow("index.html", nil)
```
If there is an error nothing happens as I don't know how you want to handle errors: panic or return error to the developer and let him handle it.

File: lib/bindings/window/window.go [hunk 1]
```diff
@@ -27,9 +29,29 @@ type Window struct {
...
-func NewWindow(url string, sess *session.Session) *Window {
+func checkUrl(s string) (string, error) {
+	u, err := url.Parse(s)
+	if err != nil {
+		return s, err
+	}
+	if u.Scheme == "" {
+		p, err := filepath.Abs(s)
+		if err != nil {
+			return p, err
```
Comment (cbandy) [inline lib/bindings/window/window.go]: Here you return the result of the failing call, but a few lines up you return the argument to the failing call. Why the difference?

**Comment by francisbouvier** [comment]
@cbandy 

You're right, it makes more sense to return the argument of the failing call.

**Comment by miketheprogrammer** [comment]
Looks good, thanks for so many PR's lately, I have been transitioning Jobs and also got engaged. With that said I have not had much time if any to maintain this project.

Also @francisbouvier if you have any experience with C or C++ please let me know. 
We would in the next few months like to revive the Thrust project by basing more closely off of Atom-Shell.
That way we can fly on the wings of Atom-Shell, but also provide the same Polyglot experience that Thrust provides.

**Comment by francisbouvier** [comment]
@miketheprogrammer 
Thanks and congrats.

Unfortunately I have too few experience in C or C++ to help with Thrust core, but I will be happy to be part of the discussion regarding the direction of the project.
I can help with the bindings in Python, Go and Javascript.

**Pull request merged**